pinot-core/src/main/java/org/apache/pinot/core/util/trace/ContinuousJfrStarter.java

```diff
@@ -0,0 +1,164 @@
+  ///
+  /// This is only used if toDisk is true.
+  /// The value is in bytes.
+  /// The default value is 200MB.
+  ///
+  /// @see #MAX_AGE
+  public static final String MAX_SIZE = "maxSize";
+  public static final int DEFAULT_MAX_SIZE = 200 * 1024 * 1024;
+  /// Key that controls the maximum age of the recording file.
+  /// Once the file reaches this age, older events will be discarded.
+  /// If both maxSize and maxAge are set, the recording will be discarded when either condition is met.
+  ///
+  /// This is only used if toDisk is true.
+  /// The value is a duration string, as defined by the Duration class.
+  /// The default value is 1 day (P1D).
+  ///
+  /// @see #MAX_SIZE
+  public static final String MAX_AGE = "maxAge";
+  public static final String DEFAULT_MAX_AGE = "P1D";
+
+  /// A flag to track whether the JFR recording has been started.
+  /// This is specially useful for testing and quickstarts, where servers, brokers and other components are executed
+  /// in the same JVM.
+  private static boolean _started = false;
+
+  private ContinuousJfrStarter() {
+  }
+
+  public synchronized static void init(PinotConfiguration config) {
+    PinotConfiguration subset = config.subset(CommonConstants.JFR);
+
+    if (!subset.getProperty(ENABLED, DEFAULT_ENABLED)) {
+      return;
+    }
+    if (_started) {
+      return;
+    }
+
+    Path directory = Path.of(subset.getProperty(DIRECTORY, Paths.get(".").toString()));
+
+    Recording recording;
+    String jfrConfName = subset.getProperty(CONFIGURATION, DEFAULT_CONFIGURATION);
+    try {
+      Configuration configuration = Configuration.getConfiguration(jfrConfName);
+      recording = new Recording(configuration);
+    } catch (ParseException e) {
+      throw new RuntimeException("Failed to parse JFR configuration '" + jfrConfName + "'", e);
+    } catch (IOException e) {
+      throw new UncheckedIOException("Failed to read JFR configuration '" + jfrConfName + "'", e);
+    }
+    boolean dumpOnExit = subset.getProperty(DUMP_ON_EXIT, DEFAULT_DUMP_ON_EXIT);
+    recording.setDumpOnExit(dumpOnExit);
+    if (dumpOnExit) {
+      try {
+        String timestamp = LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyy-MM-dd_HH-mm-ss"));
+        String filename = "recording-" + timestamp + ".jfr";
+        Path recordingPath = directory.resolve(filename);
+        boolean newFile = recordingPath.toFile().createNewFile();
+        if (!newFile) {
+          throw new RuntimeException("Failed to create new file: " + recordingPath);
+        }
+        recording.setDestination(recordingPath);
+      } catch (IOException e) {
+        throw new UncheckedIOException("Failed to create new recording file", e);
+      }
+    }
+
+    try {
+      recording.setName(subset.getProperty(NAME, DEFAULT_NAME));
+      boolean toDisk = subset.getProperty(TO_DISK, DEFAULT_TO_DISK);
+      if (toDisk) {
```

Comment:
Do we not need to set the file name on disk here? Where is it written to by default?

Reply:
The JVM decides the file used by JFR